### ISSUE TITLE
stan_model error

### ISSUE THREAD

**Issue by rtrangucci** [stan-dev/rstan#382]
#### Summary:

calling `stan_model` on a `.stan` file results in an an error.

Of note: I can't reproduce this error on my laptop, this is only reproducible on an RHEL server.

#### Description:

If I call stan_model on a stan file, I get:

```
starting worker pid=1470 on localhost:11117 at 12:46:59.214
Error in checkForRemoteErrors(lapply(cl, recvResult)) : 
  one node produced an error: unused argument (allow_undefined = allow_undefined)
```

#### Reproducible Steps:

Here is the `stan` file I'm trying to compile, called `test.stan`:

```stan
data {
}
parameters {
  real y;
}
model {
  y ~ normal(0, 1);
}
```

Calling:

```
library(rstan)
> model <- stan_model('test.stan')
```

leads to the error. 

#### Current Output:

The only output is:

```
starting worker pid=1649 on localhost:11117 at 12:51:03.204
Error in checkForRemoteErrors(lapply(cl, recvResult)) : 
  one node produced an error: unused argument (allow_undefined = allow_undefined)
```

#### Expected Output:

A compiled model.

#### RStan Version:

2.14.1
rstan (Version 2.14.1, packaged: 2016-12-28 14:55:41 UTC, GitRev: 5fa1e80eb817)

#### R Version:

[1] "R version 3.2.3 (2015-12-10)"

#### Session info:

```
R version 3.2.3 (2015-12-10)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server 7.2 (Maipo)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rstan_2.14.1         StanHeaders_2.14.0-1 ggplot2_2.2.1       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.7      assertthat_0.1   grid_3.2.3       plyr_1.8.3       gtable_0.2.0     stats4_3.2.3     scales_0.4.1     lazyeval_0.2.0   tools_3.2.3     
[10] munsell_0.4.3    parallel_3.2.3   inline_0.3.14    colorspace_1.2-6 gridExtra_2.2.1  tibble_1.2 
```

**Comment by jgabry** [comment]
Can you force it to run with 1 core? Often parts of error messages are
swallowed when running in parallel.

**Comment by rtrangucci** [comment]
I've tried both:

```R
library(rstan)
options(mc.cores = 1)
model <- stan_model('test.stan')
```

and

```R
library(rstan)
model <- stan('test.stan', cores=1)
```

But I still get

```
starting worker pid=5012 on localhost:11240 at 13:51:03.638
Error in checkForRemoteErrors(lapply(cl, recvResult)) : 
  one node produced an error: unused argument (allow_undefined = allow_undefined)
```

for each call.

**Comment by bob-carpenter** [comment]
It works fine for me in the R GUI on Mac OS X with
RStan 2.14.1.

- Bob

**Comment by bgoodri** [comment]
Rob, I think the rstan package could incompletely installed on RHEL. Can
you do remove.packages("rstan"), make double sure the whole directory is
gone on the disk, and reinstall rstan (making sure not to load any old
stanfit objects at startup that are often buried in a .RData file)

**Comment by rtrangucci** [comment]
Thanks, @bgoodri we'll try this and report back. I'm going to close the issue, it does seem to be localized to our machine

**Comment by kylebaron** [comment]
Same thing on fresh install for stan and all dependencies.

```r
data {
}
parameters {
  real y;
}
model {
  y ~ normal(0, 1);
}


```
```r
    knitr::opts_chunk$set(comment=".")

    .libPaths('lib')
    library(rstan)

    . Loading required package: ggplot2
    . Loading required package: StanHeaders
    . rstan (Version 2.14.1, packaged: 2016-12-28 14:55:41 UTC, GitRev: 5fa1e80eb817)
    . For execution on a local, multicore CPU with excess RAM we recommend calling
    . rstan_options(auto_write = TRUE)
    . options(mc.cores = parallel::detectCores())

    options(mc.cores = 1)

    model <- try(stan_model('test.stan'))

    model

    . [1] "Error in checkForRemoteErrors(lapply(cl, recvResult)) : \n  one node produced an error: unused argument (allow_undefined = allow_undefined)\n"
    . attr(,"class")
    . [1] "try-error"
    . attr(,"condition")
    . <simpleError in checkForRemoteErrors(lapply(cl, recvResult)): one node produced an error: unused argument (allow_undefined = allow_undefined)>

    sessionInfo()

    . R version 3.2.3 (2015-12-10)
    . Platform: x86_64-pc-linux-gnu (64-bit)
    . Running under: Ubuntu precise (12.04.2 LTS)
    . 
    . locale:
    .  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
    .  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
    .  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
    .  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
    .  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
    . [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
    . 
    . attached base packages:
    . [1] stats     graphics  grDevices utils     datasets  methods   base     
    . 
    . other attached packages:
    . [1] rstan_2.14.1         StanHeaders_2.14.0-1 ggplot2_2.1.0       
    . 
    . loaded via a namespace (and not attached):
    .  [1] Rcpp_0.12.3      digest_0.6.8     grid_3.2.3       plyr_1.8.3      
    .  [5] gtable_0.2.0     stats4_3.2.3     magrittr_1.5     evaluate_0.8    
    .  [9] scales_0.4.0     stringi_1.0-1    rmarkdown_0.9.5  tools_3.2.3     
    . [13] stringr_1.0.0    munsell_0.4.3    
```

**Comment by billgillespie** [comment]
I think this may be unique to rstan 2.14.1 running in RStudio on linux. The code that prompts the error appears to be the following if-block in stanc:

>   if (.Platform$OS.type == "unix" && R.Version()$os == "linux-gnu" &&
>       identical(Sys.getenv("RSTUDIO"), "1")) {
>     cl <- parallel::makePSOCKcluster(1L, outfile = "")
>     on.exit(parallel::stopCluster(cl))
>     parallel::clusterEvalQ(cl, Sys.setenv(RSTUDIO = 0))
>     parallel::clusterExport(cl, c("obfuscate_model_name",
>                                   "allow_undefined"), environment())
>     if (missing(file)) {
>       parallel::clusterExport(cl, "model_code", environment())
>       out <- parallel::clusterEvalQ(cl, stanc(model_code = model_code,
>                                                      obfuscate_model_name = obfuscate_model_name,
>                                                      allow_undefined = allow_undefined))
>     }
>     else {
>       parallel::clusterExport(cl, "file", environment())
>       out <- parallel::clusterEvalQ(cl, stanc(file,
>                                                      obfuscate_model_name = obfuscate_model_name,
>                                                      allow_undefined = allow_undefined))
>     }
>   }
> 

In particular it happens on the clusterEvalQ call.

The code runs fine on the same Linux system if I run it in batch (R CMD BATCH). It also runs in the Rstudio environment when I delete the offending if-block.

**Comment by nicokist** [comment]
I have also encountered this error on Rstudio Server in Linux.
```{r}
scode <- "
+ parameters {
+ real y[2]; 
+ } 
+ model {
+ y[1] ~ normal(0, 1);
+ y[2] ~ double_exponential(0, 2);
+ } 
+ "
> fit1 <- stan(model_code = scode, iter = 10, verbose = FALSE) 
starting worker pid=15380 on localhost:11597 at 13:29:56.313
Error in checkForRemoteErrors(lapply(cl, recvResult)) : 
  one node produced an error: unused argument (allow_undefined = allow_undefined)
```

```
> sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 8 (jessie)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rstan_2.14.2         StanHeaders_2.14.0-1 ggplot2_2.2.1       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.8      assertthat_0.1   grid_3.3.2       plyr_1.8.4       gtable_0.2.0    
 [6] stats4_3.3.2     scales_0.4.1     lazyeval_0.2.0   tools_3.3.2      munsell_0.4.3   
[11] parallel_3.3.2   inline_0.3.14    colorspace_1.3-0 gridExtra_2.2.1  tibble_1.2   
```

**Comment by aornugent** [comment]
This error persists in Stan 2.15 on RStudio Server Linux. Can be reproduced quickly with the latest docker-stan install (https://github.com/jrnold/docker-stan).
```R
> testfit <- stan(m, data=data_list, iter=1, chains = 1, init_r = 1)
starting worker pid=910 on localhost:11093 at 23:54:40.766

 *** caught illegal operation ***
address 0x7f3e3271ab0c, cause 'illegal operand'

Traceback:
 1: .Call("CPP_stanc280", model_code, model_cppname, allow_undefined)
 2: rstan::stanc(file, obfuscate_model_name = obfuscate_model_name,     allow_undefined = allow_undefined)
 3: (function (expr, envir = parent.frame(), enclos = if (is.list(envir) ||     is.pairlist(envir)) parent.frame() else baseenv()) .Internal(eval(expr, envir, enclos)))(quote(rstan::stanc(file,     obfuscate_model_name = obfuscate_model_name, allow_undefined = allow_undefined)),     env = quote(<environment>))
 4: (function (expr, envir = parent.frame(), enclos = if (is.list(envir) ||     is.pairlist(envir)) parent.frame() else baseenv()) .Internal(eval(expr, envir, enclos)))(quote(rstan::stanc(file,     obfuscate_model_name = obfuscate_model_name, allow_undefined = allow_undefined)),     env = quote(<environment>))
 5: do.call(msg$data$fun, msg$data$args, quote = TRUE)
 6: doTryCatch(return(expr), name, parentenv, handler)
 7: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 8: tryCatchList(expr, classes, parentenv, handlers)
 9: tryCatch(do.call(msg$data$fun, msg$data$args, quote = TRUE),     error = handler)
10: doTryCatch(return(expr), name, parentenv, handler)
11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
12: tryCatchList(expr, classes, parentenv, handlers)
13: tryCatch({    msg <- recvData(master)    if (msg$type == "DONE") {        closeNode(master)        break    }    else if (msg$type == "EXEC") {        success <- TRUE        handler <- function(e) {            success <<- FALSE            structure(conditionMessage(e), class = c("snow-try-error",                 "try-error"))        }        t1 <- proc.time()        value <- tryCatch(do.call(msg$data$fun, msg$data$args,             quote = TRUE), error = handler)        t2 <- proc.time()        value <- list(type = "VALUE", value = value, success = success,             time = t2 - t1, tag = msg$data$tag)        msg <- NULL        sendData(master, value)        value <- NULL    }}, interrupt = function(e) NULL)
14: slaveLoop(makeSOCKmaster(master, port, timeout, useXDR))
15: parallel:::.slaveRSOCK()
An irrecoverable exception occurred. R is aborting now ...
Error in unserialize(node$con) : error reading from connection 
````
```R
> sessionInfo()
R version 3.4.0 (2017-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 8 (jessie)

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblasp-r0.2.12.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rstan_2.15.1         StanHeaders_2.15.0-1 dplyr_0.5.0         
 [4] purrr_0.2.2.2        readr_1.1.0          tidyr_0.6.2         
 [7] tibble_1.3.0         ggplot2_2.2.1        tidyverse_1.1.1     
[10] MASS_7.3-47         

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.10     compiler_3.4.0   cellranger_1.1.0 plyr_1.8.4      
 [5] forcats_0.2.0    tools_3.4.0      jsonlite_1.4     lubridate_1.6.0 
 [9] gtable_0.2.0     nlme_3.1-131     lattice_0.20-35  psych_1.7.5     
[13] DBI_0.6-1        parallel_3.4.0   haven_1.0.0      gridExtra_2.2.1 
[17] xml2_1.1.1       stringr_1.2.0    httr_1.2.1       hms_0.3         
[21] stats4_3.4.0     grid_3.4.0       inline_0.3.14    R6_2.2.1        
[25] readxl_1.0.0     foreign_0.8-68   reshape2_1.4.2   modelr_0.1.0    
[29] magrittr_1.5     scales_0.4.1     assertthat_0.2.0 mnormt_1.5-5    
[33] rvest_0.3.2      colorspace_1.3-2 stringi_1.1.5    lazyeval_0.2.0  
[37] munsell_0.4.3    broom_0.4.2    
```

**Comment by bob-carpenter** [comment]
@bgoodri --- any idea what's going on with this R/Linux combo?

**Comment by billgillespie** [comment]
In stanc try replacing parallel::makePSOCKcluster with parallel::makeForkCluster. That works in our environment using RStudio Server on Linux.

**Comment by bgoodri** [comment]
On Linux, it shouldn't even be calling makePSOCKcluster() unless you call
it from RStudio.

**Comment by billgillespie** [comment]
I am referring to the situation where stan() is called from RStudio Server on Linux. I'm guessing something prevents passing of needed environment info when makePSOCKcluster is used in our system.

**Comment by aornugent** [comment]
Replacing `parallel::makePSOCKcluster` with `parallel::makeForkCluster` using `trace()` results in:

```R
Error in unserialize(node$con) : error reading from connection 
11. unserialize(node$con) 
10. recvData.SOCK0node(con) 
9. recvData(con) 
8. FUN(X[[i]], ...) 
7. lapply(cl, recvResult) 
6. checkForRemoteErrors(lapply(cl, recvResult)) 
5. clusterCall(cl, eval, substitute(expr), env = .GlobalEnv) 
4. parallel::clusterEvalQ(cl, rstan::stanc(file, obfuscate_model_name = obfuscate_model_name,    allow_undefined = allow_undefined)) at <tmp>#19
3. stanc(file = file, model_code = model_code, model_name = model_name, verbose = verbose, obfuscate_model_name = obfuscate_model_name, allow_undefined = allow_undefined) 
2. stan_model(file, model_name = model_name, model_code = model_code, stanc_ret = NULL, boost_lib = boost_lib, eigen_lib = eigen_lib,  save_dso = save_dso, verbose = verbose) 
1. stan(m, data = data_list, iter = 1, chains = 1, init_r = 1) 
```

Any further suggestions?

**Comment by bgoodri** [comment]
Yes, on RStudio it should call makePSOCKcluster() but otherwise on Linux
servers it is just calling mclapply(). I think there could be security
settings or something that prevent makePSOCKcluster() from working.

**Comment by bgoodri** [comment]
Does it work on your server if you call your R code from the terminal
rather than from RStudio?

**Comment by aornugent** [comment]
R from the terminal results in:

```R
 testfit <- stan(m, data=data_list, iter=1, chains = 1, init_r = 1)

 *** caught illegal operation ***
address 0x7f070f723b0c, cause 'illegal operand'

Traceback:
 1: .Call("CPP_stanc280", model_code, model_cppname, allow_undefined)
 2: stanc(file = file, model_code = model_code, model_name = model_name, verbose = verbose, obfuscate_model_name = obfuscate_model_name, allow_undefined = allow_undefined)
 3: stan_model(file, model_name = model_name, model_code = model_code, stanc_ret = NULL, boost_lib = boost_lib, eigen_lib = eigen_lib, save_dso = save_dso, verbose = verbose)
 4: stan(m, data = data_list, iter = 1, chains = 1, init_r = 1)

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace
```

**Comment by bgoodri** [comment]
Maybe try with -O2 instead of -O3 in ~/.R/Makevars? Or the reverse if you
are doing -O2 or -Os now.

**Comment by aornugent** [comment]
Tweaking jrnold's dockerfile to use the -O2 tags causes the build to hang when compiling on several different computers. Key excerpt below, the full logs can be seen here: https://hub.docker.com/r/aornugent/docker-stan/builds/bbunfihmsgcvbuge3cyepna/ 

```
* installing *source* package ‘rstan’ ...
** package ‘rstan’ successfully unpacked and MD5 sums checked
** libs
g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c chains.cpp -o chains.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.cpp -o init.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__ast_def.cpp -o lang__ast_def.o

In file included from /usr/local/lib/R/site-library/StanHeaders/include/src/stan/lang/ast_def.cpp:112:0,
                 from lang__ast_def.cpp:18:
/usr/local/lib/R/site-library/StanHeaders/include/src/stan/lang/ast/sigs/function_signatures_def.hpp: In constructor ‘stan::lang::function_signatures::function_signatures()’:
/usr/local/lib/R/site-library/StanHeaders/include/src/stan/lang/ast/sigs/function_signatures_def.hpp:411:5: note: variable tracking size limit exceeded with -fvar-tracking-assignments, retrying without
     function_signatures::function_signatures() {
     ^

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__bare_type_grammar_inst.cpp -o lang__grammars__bare_type_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__expression07_grammar_inst.cpp -o lang__grammars__expression07_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__expression_grammar_inst.cpp -o lang__grammars__expression_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__functions_grammar_inst.cpp -o lang__grammars__functions_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__indexes_grammar_inst.cpp -o lang__grammars__indexes_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__program_grammar_inst.cpp -o lang__grammars__program_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__semantic_actions.cpp -o lang__grammars__semantic_actions.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__statement_2_grammar_inst.cpp -o lang__grammars__statement_2_grammar_inst.o

g++  -I/usr/local/lib/R/include -DNDEBUG -I"../inst/include" -I"../inst/include/boost_not_in_BH" -I"`"/usr/local/lib/R/bin/Rscript" --vanilla -e "cat(system.file('include', 'src', package = 'StanHeaders'))"`" -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppEigen/include" -I"/usr/local/lib/R/site-library/BH/include" -I"/usr/local/lib/R/site-library/StanHeaders/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lang__grammars__statement_grammar_inst.cpp -o lang__grammars__statement_grammar_inst.o
...[hangs]
```

**Comment by apteroti** [comment]
I'm so joyful to see the issue hasn't been fixed yet (stan version: 2.26)! True R style magic, good job devs!
